From 8a5d1e5a34e7e3d3ee001904914e02d5983e2457 Mon Sep 17 00:00:00 2001
From: Christopher Georg <christopher.georg@sr-travel.de>
Date: Mon, 14 Oct 2024 12:55:36 +0200
Subject: [PATCH] fix: deprecation implicit nullable PHP 8.4

---
 src/Differ.php | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/Differ.php b/src/Differ.php
index 583d2da7..a34dc56d 100644
--- a/src/Differ.php
+++ b/src/Differ.php
@@ -82,7 +82,7 @@ public function __construct($outputBuilder = null)
      * @param array|string $from
      * @param array|string $to
      */
-    public function diff($from, $to, LongestCommonSubsequenceCalculator $lcs = null): string
+    public function diff($from, $to, ?LongestCommonSubsequenceCalculator $lcs = null): string
     {
         $diff = $this->diffToArray(
             $this->normalizeDiffInput($from),
@@ -106,9 +106,9 @@ public function diff($from, $to, LongestCommonSubsequenceCalculator $lcs = null)
      *
      * @param array|string                       $from
      * @param array|string                       $to
-     * @param LongestCommonSubsequenceCalculator $lcs
+     * @param ?LongestCommonSubsequenceCalculator $lcs
      */
-    public function diffToArray($from, $to, LongestCommonSubsequenceCalculator $lcs = null): array
+    public function diffToArray($from, $to, ?LongestCommonSubsequenceCalculator $lcs = null): array
     {
         if (is_string($from)) {
             $from = $this->splitStringByLines($from);
